### PR TITLE
Customer confirm pickup delivery

### LINKED ISSUE #4: As a customer I want to confirm pickup/delivery
For each customer that has an active order on their profile:
- confirm pickup by a driver and change status to "expected"
- confirm delivery when the order is received by the customer, change status to "collected"

### PR DESCRIPTION
**Pull request by Irysek707**
Fix to the previous PR with commets taken into consideration.

In this PR I added:
1. Buttons to go back to the Main Page from all the user pages.
2. Buttons to go back to Customer Page from Order Preview Page.
3. Button and functionality to confirm Pickup by a Customer and change the order status to Expected (when Awaiting Pickup).
4. Button and functionality to confirm Delivery by Customer and change the order status to Collected (when Expecting).

image
image
image

### CODE REVIEW

File: Haulage/View/OrderPreview.xaml.cs [hunk 1]
```diff
@@ -3,131 +3,108 @@
...
+                ErrorMessage.Text = "Please select a valid date to pickup";
+            }
         }
-    }
 
-    private async void BackToCustomerPageBtn_Clicked(object sender, EventArgs e)
-    {
-        // Navigate back to the customer page
-        App.Current.MainPage = new NavigationPage(new CustomerPage(customer)); // Assuming CustomerPage is the page for customer overview
+        private void RefreshPage()
```
Comment (RandomUni) [inline Haulage/View/OrderPreview.xaml.cs:100]: Good job on adding the page refresh as separate method

File: Haulage/View/OrderPreview.xaml.cs [hunk 2]
```diff
@@ -3,131 +3,108 @@
...
-        try
-        {
-            // Change the status to COLLECTED
-            order.Status = Model.Constants.Status.COLLECTED;
-            // Update the order in the database
-            DB.connection.Update(order);
+            Status.Text = "Current order status " + orderToPreview.Status;
 
-            await DisplayAlert("Success", "Delivery confirmed. Status updated to COLLECTED.", "OK");
+            // Set button visibility based on order status
```
Comment (RandomUni) [inline Haulage/View/OrderPreview.xaml.cs:34]: Good Job on adding extra validation

**Review by RandomUni — APPROVED** [review]

File: Haulage/Model/CustomerOrder.cs [hunk 3]
```diff
@@ -51,15 +51,27 @@ public void AddManifest(Manifest manifest)
...
+        {
+            this.status = Status.EXPECTED;
+            DB.connection.Update(this);
+        }
+
+        public void ConfirmDelivery()
+        {
+            this.status = Status.COLLECTED;
+            DB.connection.Update(this);
         }
```
Comment (Domkw1) [inline Haulage/Model/CustomerOrder.cs:75]: The methods are well-structured to confirm the change in order status

**Review by Domkw1 — APPROVED** [review]

File: Haulage/View/OrderPreview.xaml.cs [hunk 4]
```diff
@@ -3,131 +3,108 @@
...
-        {
-            // Change the status to EXPECTED
-            order.Status = Model.Constants.Status.EXPECTED;
-            // Update the order in the database
-            DB.connection.Update(order);
-
-            await DisplayAlert("Success", "Pickup confirmed. Status updated to EXPECTED.", "OK");
+        private CustomerOrder order;
+        private Customer customer;
+        private CustomerController controller;
```
Comment (Domkw1) [inline Haulage/View/OrderPreview.xaml.cs:13]: Customer Properties added in

**Pull request merged**